### DISCUSSION TITLE
Guidance Request: Transitioning from Docker Compose (Portainer Stack with Separate MariaDB) to Kubernetes/Helm Deployment for Firefly III

### DISCUSSION THREAD

**Discussion by ameeno** [firefly-iii/firefly-iii#8470] [Firefly III]
Hello,

As the subject indicates, I'm exploring a migration of my current Firefly III setup from a Portainer stack, which includes a separate MariaDB instance located on the same Docker host, to a Kubernetes environment utilizing Helm charts. Additionally, this setup incorporates a standalone Redis cache.

I'm contemplating the most efficient approach to shift to Kubernetes. My initial thought is to perform a MySQL dump of the database and then integrate this data with the Kubernetes deployment. However, I'm considering whether a data export from the current setup followed by an import into the Kubernetes installation would be more effective.

Could you advise on the best practice for this migration? What steps would you recommend?

Thank you for your guidance.

**Comment by JC5** [comment]
MySQL dump. Don't use the import/export function for this.

**Comment by ameeno** [comment]
Hi JC5,

Thanks for the suggestion. I tried it but encountered issues importing the MYSQL dump into PostgreSQL. However, I managed to migrate the database using pgloader. One issue was pgloader migrated the data under the original MySQL schema name, but Firefly 3 needs it under "public". Fixed it by renaming the schema to "public" after dropping the existing one. Adjusted Firefly's settings to use PostgreSQL, and it worked. Now, I'm ready to pgdump and move to the k8s deployment. Cheers!

**Reply by JC5** [reply]
Nice work, thanks for the update!
